### PR TITLE
Add convert function for string and bool

### PR DESCRIPTION
**Pull request by oovm**
I think `try_*` does not follow the rust naming convention, should use `as_*`

### CODE REVIEW

**Comment by ConnorGray** [comment]
@oovm Thank you for making these contributions!

I originally chose to use the `try_*` because I think these methods are more like `TryFrom/TryInto` conversions than `as_*` functions (which I think don't typically fail?). Though `as_*` does more clearly suggest that a conversion is happening. Perhaps `try_as_*`?

**Review by ConnorGray — CHANGES_REQUESTED** [review]
Thank you for submitting this pull request! :) 

New convenience functions like this will be a big help to reducing some of the verbosity that is currently required when working with expressions.

I've left a few comments about a couple of conversions that I think we might want to approach in a different way, perhaps in a future PR.

File: src/lib.rs [hunk 1]
```diff
@@ -485,12 +505,47 @@ impl From<Normal> for Expr {
     }
 }
 
-impl From<&str> for Expr {
-    fn from(str: &str) -> Expr {
-        Expr::string(str)
+impl From<()> for Expr {
```
[the remark this answers is not in the document]
Comment (oovm) [reply src/lib.rs:508]: You are right, I will remove `()` and `Option` both

**Comment by oovm** [comment]
rebase complete

**Review by ConnorGray — CHANGES_REQUESTED** [review]
I apologize, I wrote out this second round of review comments several weeks ago, but apparently forgot to finish submitting them to the PR.

File: src/lib.rs [hunk 2]
```diff
@@ -287,6 +274,23 @@ impl Expr {
...
+    /// use wolfram_expr::Expr;
+    ///
+    /// let assoc = Expr::association(vec![
+    ///     Expr::rule(Expr::from("a"), Expr::from(1)),
+    ///     Expr::rule_delayed(Expr::from("b"), Expr::from(2)),
+    /// ]);
+    /// ```
+    pub fn association(elements: Vec<Expr>) -> Expr {
+        Expr::normal(Symbol::new("System`Association"), elements)
+    }
```
[the remark this answers is not in the document]
Comment (oovm) [reply src/lib.rs:293]: It is actually possible to design an auxiliary structure pub struct `Association {wrap: IndexMap}`

[the remark this answers is not in the document]
Comment (oovm) [reply src/lib.rs:293]: A lot of design is needed here, I'll implement this in the next pr if appropriate.

**Comment by ConnorGray** [comment]
@oovm Thank you again for the PR!

**Pull request merged**